src/common/meta/src/reconciliation/utils.rs

```diff
@@ -433,6 +433,105 @@ pub(crate) async fn validate_table_id_and_name(
     Ok(())
 }
 
+/// Resolves the column metadata for a logical table by selecting the column metadata
+/// from the region with the maximum number of columns.
+///
+/// # Panics
+/// Panics if `region_metadatas` is empty.
+///
+/// TODO(weny): add tests
+pub(crate) fn resolve_column_metadatas_for_logical_table(
+    region_metadatas: &[RegionMetadata],
+    table_info: &RawTableInfo,
+) -> Result<Vec<ColumnMetadata>> {
+    let is_same_table = region_metadatas
+        .windows(2)
+        .all(|w| w[0].region_id.table_id() == w[1].region_id.table_id());
+
+    ensure!(
+        is_same_table,
+        UnexpectedSnafu {
+            err_msg: "Region metadatas are not from the same table"
+        }
+    );
+
+    let region_metadata = region_metadatas
+        .iter()
+        .max_by_key(|r| r.column_metadatas.len())
+        .unwrap();
+
+    let is_invariant_preserved = check_column_metadatas_invariants_for_logical_table(
+        &region_metadata.column_metadatas,
+        table_info,
+    );
+    ensure!(
+        is_invariant_preserved,
+        UnexpectedSnafu {
+            err_msg: format!(
+                "Column metadata invariants violated for region {}. Resolved column metadata: {:?}, region column metadata: {:?}",
+                region_metadata.region_id,
+                region_metadata.column_metadatas.iter().map(ColumnMetadataDisplay).collect::<Vec<_>>(),
+                region_metadata.column_metadatas.iter().map(ColumnMetadataDisplay).collect::<Vec<_>>()
```

Comment:
print the same vec twice?

Reply:
The function isn't used, removed in [d35d6ad](https://github.com/GreptimeTeam/greptimedb/pull/6588/commits/d35d6ad279510d7df8203d9c053da7bfd783379f)